### PR TITLE
Update Issue/PR Template text

### PR DESCRIPTION
**Pull request by WaLLy3K**
**By submitting this pull request, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

9

---

* Standardise both templates for our needs
* Provided directions for each heading
* Provide 'quoted section' text block of requests to the user

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,19 +1,32 @@
-**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**
+**By submitting this pull request, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`
 
-***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***
+`{Please ensure that your pull request is for the 'development' branch!}`
 
 - [] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
-- [] I have written tests and verified that they fail without my change.
+- [] I have performed relevant testing to verify that they work with my change.
```
Comment (Mcat12) [inline .github/PULL_REQUEST_TEMPLATE.md]: The `they` reference can be confusing. Maybe rephrase it to: `I have performed relevant testing and, if applicable, I have written tests to verify the feature works in the future`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
-- [] I have Signed Off all commits. (`git commit --signoff`)
+- [] I give this submission freely and claim no ownership.
+- [] It is compatible with the EUPL 1.2 license.
+- [] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))
 
-***Please explain what you have done and wish to accomplish with this Pull Request***
+---
 
-1. What does this change do, exactly?
+**What does this PR aim to fix?:**
```
Comment (Mcat12) [inline .github/PULL_REQUEST_TEMPLATE.md]: It may be a feature PR, in which case this text should read more like: `What does this PR aim to accomplish?`
The following prompts should also be changed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (WaLLy3K) [inline .github/PULL_REQUEST_TEMPLATE.md]: I read "I have written tests to verify the feature works in the future" and I think that I'm required to modify the `python` tests to ensure that it passes Travis. Is this what we're asking of users too, or something else?

Comment (Mcat12) [inline .github/PULL_REQUEST_TEMPLATE.md]: Post discussion: we shouldn't ask users to make tests as it's a relatively big undertaking when you aren't familiar with the CI, and they probably won't be great tests.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+**How familiar are you with the the source code relevant to this issue?:**
 
-_{replace this text with a number from 1 to 10, with 1 being not familiar, and 10 being very familiar}_
+`{Replace this with a number from 1 to 10. 1 being not familiar, and 10 being very familiar}`
 
 ---
-**[BUG REPORT | OTHER]:**
+**Expected behaviour:**
 
-Please [submit your feature request here](https://discourse.pi-hole.net/c/feature-requests), so it is votable by the community.  It's also easier for us to track.
```
Comment (PromoFaux) [inline .github/ISSUE_TEMPLATE.md]: How about something more along the lines of:

**Feature Request**
- If this is a feature request, please do not submit this as an issue here. Instead post it to [our discourse forums](https://discourse.pi-hole.net/c/feature-requests) so that it can be better tracked, and voted on by the community

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (dschaper) [reply .github/PULL_REQUEST_TEMPLATE.md]: `[] I have tested my proposed changes, and have included unit tests where possible.`

Make the wording less verbose and just put it plainly.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,19 +1,32 @@
-**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**
+**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`
 
-***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***
+`{Please ensure that your pull request is for the 'development' branch!}`
 
 - [] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
-- [] I have written tests and verified that they fail without my change.
+- [] I have performed relevant testing to verify that they work with my change.
 - [] I have squashed any insignificant commits.
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: `[] I have squashed any insignificant commits with git rebase.` 

Give them a hint as to how to do the action.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,19 +1,32 @@
...
 - [] This change has comments for package types, values, functions, and non-obvious lines of code.
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: `[] I have commented my proposed changes in the code.`

The original was just a copy of a template, way too complex verbiage for most people.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,19 +1,32 @@
...
 - [] I am willing to help maintain this change if there are issues with it later.
-- [] I give this submission freely and claim no ownership. It is compatible with the EUPL 1.2 license.
-- [] I have Signed Off all commits. (`git commit --signoff`)
+- [] I give this submission freely and claim no ownership.
+- [] It is compatible with the EUPL 1.2 license.
+- [] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: `[] I have only made one major change in my proposed changes.`

Try to head off big PR's that change a lot of code in one shot. We can cherry-pick, but better to just not gold plate to being with.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,19 +1,32 @@
...
+
+**What documentation changes (if any) are needed to support this PR?:**
+
+`{A detailed list of any necessary changes}`
+
+> * `{Please delete this quoted section when opening your pull request}`
+> * You must follow the template instructions. Failure to do so will result in your issue being closed.
+> * Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.
+> * Detail helps us understand an issue quicker, but please ensure it's relevant.
+> * _This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can probably lose this line as we've diverged pretty well from the origin.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -1,19 +1,32 @@
-**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**
+**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`
 
-***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***
+`{Please ensure that your pull request is for the 'development' branch!}`
 
 - [] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: `[] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md) and this entire template.`

Another hint to look at the end of the template for instructions.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -1,19 +1,32 @@
...
+- [] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
+- [] I have made only one major change in my proposed changes.
+- [] I have commented my proposed changes within the code.
+- [] I have tested my proposed changes, and have included unit tests where possible.
 - [] I am willing to help maintain this change if there are issues with it later.
-- [] I give this submission freely and claim no ownership. It is compatible with the EUPL 1.2 license.
-- [] I have Signed Off all commits. (`git commit --signoff`)
+- [] I give this submission freely and claim no ownership.
+- [] It is compatible with the EUPL 1.2 license.
+- [] I have squashed any insignificant commits using [`git rebase`](https://git-scm.com/docs/git-rebase).
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html may be a little more user-friendly intro to the rebase command. The docs are kind of dry to read.

**Comment by dschaper** [comment]
Looking great, just one last suggestion on it.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -1,19 +1,32 @@
...
-- [] This change has comments for package types, values, functions, and non-obvious lines of code.
+- [] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
+- [] I have made only one major change in my proposed changes.
+- [] I have commented my proposed changes within the code.
+- [] I have tested my proposed changes, and have included unit tests where possible.
 - [] I am willing to help maintain this change if there are issues with it later.
-- [] I give this submission freely and claim no ownership. It is compatible with the EUPL 1.2 license.
-- [] I have Signed Off all commits. (`git commit --signoff`)
+- [] I give this submission freely and claim no ownership.
+- [] It is compatible with the EUPL 1.2 license.
```
Comment (dschaper) [inline .github/PULL_REQUEST_TEMPLATE.md]: https://opensource.org/licenses/EUPL-1.1 Link for EUPL (Yes, it actually is 1.2, the link is correct.)
